### PR TITLE
Add documentation string

### PR DESCRIPTION
**Pull request by fstamour**
Hello, 

I added a few documentation strings.

The rest is mainly removing trailing whitespaces.
Not sure what's your opinion about that kind of changes (styling), but I figured I could just open the PR and see.

Thank you

### CODE REVIEW

File: ltk/ltk.lisp [hunk 1]
```diff
@@ -839,6 +833,7 @@ fconfigure stdout -encoding utf-8
   (throw *wish* nil))
 
 (defun send-wish (text)
+  "send text to wish process and flush if *buffer-for-atomic-output* is true"
```
Comment (fstamour) [inline ltk/ltk.lisp:836]: Added

File: ltk/ltk.lisp [hunk 2]
```diff
@@ -1058,6 +1052,7 @@ fconfigure stdout -encoding utf-8
       nil)))
 
 (defun tcldebug (something)
+  "print a sting on standard-output prefixed with \"tcl debug:\""
```
Comment (fstamour) [inline ltk/ltk.lisp:1055]: Added

File: ltk/ltk.lisp [hunk 3]
```diff
@@ -1074,8 +1069,7 @@ event to read and blocking is set to nil"
             no-event-value))))
 
 (defun read-data ()
-  "Read data from wish. Non-data events are postponed, bogus messages (eg.
-+error-strings) are ignored."
+  "read data from wish. Non-data events are postponed, bogus messages (eg. +error-strings) are ignored."
```
Comment (fstamour) [inline ltk/ltk.lisp:1072]: I just changed "Read" to "read" for consistency

File: ltk/ltk.lisp [hunk 4]
```diff
@@ -1154,23 +1146,19 @@ event to read and blocking is set to nil"
...
   |#
 
-;; basic tk object
 (defclass tkobject ()
-  ((name :accessor name :initarg :name :initform nil)
-   )
+  ((name :accessor name :initarg :name :initform nil))
   (:documentation "Base class for every Tk object"))
 
-;; basic class for all widgets
```
Comment (fstamour) [inline ltk/ltk.lisp]: I removed the comment, as there is already a documentation string that says the same thing.

File: ltk/ltk.lisp [hunk 5]
```diff
@@ -1154,23 +1146,19 @@ event to read and blocking is set to nil"
...
+(defclass widget (tkobject)
   ((master :accessor master :initarg :master :initform nil) ;; parent widget or nil
-   (widget-path :initarg :path :initform nil :accessor %widget-path)         ;; pathname to refer to the widget
-   (init-command :accessor init-command :initform nil :initarg :init-command)
-   )
+   (widget-path :initarg :path :initform nil :accessor %widget-path) ;; pathname to refer to the widget
+   (init-command :accessor init-command :initform nil :initarg :init-command))
   (:documentation "Base class for all widget types"))
 
-;; creating of the tk widget after creating the clos object
```
Comment (fstamour) [inline ltk/ltk.lisp:1171]: I removed the comment, as there is already a documentation string that says the same thing.

File: ltk/ltk.lisp [hunk 6]
```diff
@@ -1154,23 +1146,19 @@ event to read and blocking is set to nil"
...
 (defmethod initialize-instance :after ((w widget) &key)
-  (unless (name w)			; generate name if not given
+  "initialize the tk widget after the clos object is made."
```
Comment (fstamour) [inline ltk/ltk.lisp:1160]: Added

File: ltk/ltk.lisp [hunk 7]
```diff
@@ -1179,22 +1167,25 @@ event to read and blocking is set to nil"
 ;;; tcl -> lisp: puts "$x" mit \ und " escaped
 ;;;  puts [regsub {"} [regsub {\\} $x {\\\\}] {\"}]
 
-;;; call to convert untility
-(defun convert(from to)
+(defun convert (from to)
```
Comment (fstamour) [inline ltk/ltk.lisp:1170]: Removed the comment and added a docstring.

File: ltk/ltk.lisp [hunk 8]
```diff
@@ -1179,22 +1167,25 @@ event to read and blocking is set to nil"
...
+  "utility to call tcl's convert function"
   (close-process-stream (do-execute "convert" (list from to) t)))
 
 ;;; table used for callback every callback consists of a name of a widget and
 ;;; a function to call
 
+(defun format-trace (format-string &rest args)
```
Comment (fstamour) [inline ltk/ltk.lisp:1177]: I added this function, it's used just below

**Comment by herth** [comment]
Merged in with some edits.

**Comment by fstamour** [comment]
'bout time :P

**Pull request closed without merging**